jni/com_eclipsesource_v8_V8Impl.cpp

```diff
@@ -1469,3 +1472,215 @@ JNIEXPORT jlong JNICALL Java_com_eclipsesource_v8_V8__1getBuildID
   (JNIEnv *, jobject) {
   return 2;
 }
+
+/***** Serializes objects to JSON on the fly (used by code to serialize profiling information to file) *****/
+
+static void findAndReplace(string& source, const string find, const string replace)
+{
+    string::size_type i = 0;
+    while((i = source.find(find, i)) != string::npos) {
+      source.replace(i, find.length(), replace);
+      i += replace.length();
+    }
+}
+
+class JsonSerializer {
```

Comment:
There may be a third party library I could use for this, but I was trying to keep things simple as it's not a complicated class. Also, if you did know of a library, it would need to support writing the serialized json on the fly (compared to just taking in an object and spitting out a string)